### ISSUE TITLE
Range end inconsistency

### ISSUE THREAD

**Issue by esledov** [nathan-cruz77/float_range#20]
```
In [4]: list(frange(8,10, step=0.5))                                                                                    
Out[4]: [8, 8.5, 9.0, 9.5]

In [5]: list(frange(8,10, step=0.2))                                                                                    
Out[5]: [8, 8.2, 8.4, 8.6, 8.8, 9.0, 9.2, 9.4, 9.6, 9.8, 10.0]

```

Why 10 is included when step is 0.2 and not included when step is 0.5?

**Comment by gmelodie** [comment]
It seems the problem is in the criteria for stopping the iteration. Iteration stops when `__next__` raises `StopIteration`, which in our case means the `_is_empty` function returned `True`:

https://github.com/nathan-cruz77/float_range/blob/1e7118efc25b47a3ab7592fe1e96418dee3c31c6/float_range/range.py#L23-L29

Iteration stops when `start` is greater than **or equal** to `stop` if the range is increasing (`step` is positive) -- and the equivalent for decreasing ranges:

https://github.com/nathan-cruz77/float_range/blob/1e7118efc25b47a3ab7592fe1e96418dee3c31c6/float_range/range.py#L113-L120

This should work correctly (interval should not include last element), but for some reason it includes the last element sometimes. I've tested some other values:

```
>>> list(float_range.range(8,10,0.1))
[8, 8.1, 8.2, 8.3, 8.4, 8.5, 8.6, 8.7, 8.8, 8.9, 9.0, 9.1, 9.2, 9.3, 9.4, 9.5, 9.6, 9.7, 9.8, 9.9, 10.0]
```


```
>>> list(float_range.range(8,10,0.4))
[8, 8.4, 8.8, 9.2, 9.6]
```

And backwards as well:

```
>>> list(float_range.range(10,8,-0.5))
[10, 9.5, 9.0, 8.5]
```


```
>>> list(float_range.range(10,8,-0.2))
[10, 9.8, 9.6, 9.4, 9.2, 9.0, 8.8, 8.6, 8.4, 8.2, 8.0
```

I'm guessing it has something to do with the `_precision` function (maybe it's rounding numbers incorreclty idk):

https://github.com/nathan-cruz77/float_range/blob/1e7118efc25b47a3ab7592fe1e96418dee3c31c6/float_range/range.py#L99-L111

**Comment by nathan-cruz77** [comment]
Hi, @esledov. Thanks for bringing this to our attention. I've just opened a [PR](https://github.com/nathan-cruz77/float_range/pull/21) to address this issue. As soon as @gmelodie takes a look at it we can merge.

**Comment by gmelodie** [comment]
Fixed in #21
